src/Sylius/Behat/Element/Product/ShowPage/VariantsElement.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Element\Product\ShowPage;
+
+use Behat\Mink\Element\NodeElement;
+use FriendsOfBehat\PageObjectExtension\Element\Element;
+
+final class VariantsElement extends Element implements VariantsElementInterface
+{
+    public function countVariantsOnPage(): int
+    {
+        /** @var NodeElement $variants|array */
+       $variants = $this->getDocument()->findAll('css', '#variants .variant');
+
+       return sizeof($variants);
+    }
+
+    public function hasProductVariantWithCodePriceAndCurrentlyStock(string $name, string $code, string $price, string $currentlyStock): bool
+    {
+        /** @var NodeElement $variantRow */
+        $variantRows =  $this->getDocument()->findAll('css', '#variants .variant');
+
+        /** @var NodeElement $variant */
+        foreach ($variantRows as $variant) {
+            if ($this->hasProductWithGivenNameCodePriceAndCurrentStock($variant, $name, $code, $price, $currentlyStock))
```

Comment:
This line looks like a monstrous `if` statement - how about splitting it into two lines?